MergerCli/utils/BatchStatusManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace MergerCli.Utils
+{
+    internal class BatchStatusManager
+    {
+        internal class LayerStatus
```

Comment:
i don't like the you define 2 classes at the same file.
maybe try to separate this class definition to a different file